Build fails with LTO due to ODR violations

I tried to build with the following *FLAGS to optimize the build: `-flto=4 -Werror=odr -Werror=lto-type-mismatch -Werror=strict-aliasing`

Note the -Werror=* flags are used to help detect cases where the compiler tries to optimize by assuming UB cannot exist in the source code -- if it does exist, ordinarily the code would be miscompiled, and this says to make the miscompilation a fatal error.

I got this error:
```
FAILED: ja2 
: && /usr/bin/x86_64-pc-linux-gnu-g++ -march=native -fstack-protector-all -O2 -pipe -fdiagnostics-color=always -frecord-gcc-switches -flto=4 -Werror=odr -Werror=lto-type-mismatch -Werror=strict-aliasing  -Wformat -Werror=format-security -Wl,-O1 -Wl,--as-needed -flto=4 -Werror=odr -Werror=lto-type-mismatch -Werror=strict-aliasing -Wl,--defsym=__gentoo_check_ldflags__=0 CMakeFiles/ja2.dir/src/externalized/army/ArmyCompositionModel.cc.o CMakeFiles/ja2.dir/src/externalized/army/GarrisonGroupModel.cc.o CMakeFiles/ja2.dir/src/externalized/army/PatrolGroupModel.cc.o CMakeFiles/ja2.dir/src/externalized/content/ContentMercs.cc.o CMakeFiles/ja2.dir/src/externalized/content/Dialogs.cc.o CMakeFiles/ja2.dir/src/externalized/internals/enums.cc.o CMakeFiles/ja2.dir/src/externalized/json/Json.cc.o CMakeFiles/ja2.dir/src/externalized/mercs/MercProfile.cc.o CMakeFiles/ja2.dir/src/externalized/mercs/MercProfileInfo.cc.o CMakeFiles/ja2.dir/src/externalized/mercs/MERCListingModel.cc.o CMakeFiles/ja2.dir/src/externalized/mercs/RPCSmallFaceModel.cc.o CMakeFiles/ja2.dir/src/externalized/policy/DefaultGamePolicy.cc.o CMakeFiles/ja2.dir/src/externalized/policy/DefaultIMPPolicy.cc.o CMakeFiles/ja2.dir/src/externalized/policy/DefaultStrategicAIPolicy.cc.o CMakeFiles/ja2.dir/src/externalized/scripting/EnumCodeGen.cc.o CMakeFiles/ja2.dir/src/externalized/scripting/FunctionsLibrary.cc.o CMakeFiles/ja2.dir/src/externalized/scripting/ScriptingExtensionsLua.cc.o CMakeFiles/ja2.dir/src/externalized/strategic/BloodCatPlacementsModel.cc.o CMakeFiles/ja2.dir/src/externalized/strategic/BloodCatSpawnsModel.cc.o CMakeFiles/ja2.dir/src/externalized/strategic/CacheSectorsModel.cc.o CMakeFiles/ja2.dir/src/externalized/strategic/CreatureLairModel.cc.o CMakeFiles/ja2.dir/src/externalized/strategic/FactParamsModel.cc.o CMakeFiles/ja2.dir/src/externalized/strategic/MineModel.cc.o CMakeFiles/ja2.dir/src/externalized/strategic/MovementCostsModel.cc.o CMakeFiles/ja2.dir/src/externalized/strategic/NpcPlacementModel.cc.o CMakeFiles/ja2.dir/src/externalized/strategic/SamSiteModel.cc.o CMakeFiles/ja2.dir/src/externalized/strategic/SamSiteAirControlModel.cc.o CMakeFiles/ja2.dir/src/externalized/strategic/SectorLandTypes.cc.o CMakeFiles/ja2.dir/src/externalized/strategic/StrategicMapSecretModel.cc.o CMakeFiles/ja2.dir/src/externalized/strategic/TownModel.cc.o CMakeFiles/ja2.dir/src/externalized/strategic/TraversibilityMapping.cc.o CMakeFiles/ja2.dir/src/externalized/strategic/UndergroundSectorModel.cc.o CMakeFiles/ja2.dir/src/externalized/strings/EncryptedString.cc.o CMakeFiles/ja2.dir/src/externalized/strings/ItemStrings.cc.o CMakeFiles/ja2.dir/src/externalized/strings/Localization.cc.o CMakeFiles/ja2.dir/src/externalized/tileset/TilesetTileIndexModel.cc.o CMakeFiles/ja2.dir/src/externalized/tactical/MapItemReplacementModel.cc.o CMakeFiles/ja2.dir/src/externalized/tactical/NpcActionParamsModel.cc.o CMakeFiles/ja2.dir/src/externalized/ui/Easings.cc.o CMakeFiles/ja2.dir/src/externalized/AmmoTypeModel.cc.o CMakeFiles/ja2.dir/src/externalized/CalibreModel.cc.o CMakeFiles/ja2.dir/src/externalized/DealerModel.cc.o CMakeFiles/ja2.dir/src/externalized/DealerInventory.cc.o CMakeFiles/ja2.dir/src/externalized/DefaultContentManager.cc.o CMakeFiles/ja2.dir/src/externalized/GraphicModel.cc.o CMakeFiles/ja2.dir/src/externalized/InventoryGraphicsModel.cc.o CMakeFiles/ja2.dir/src/externalized/ItemModel.cc.o CMakeFiles/ja2.dir/src/externalized/JsonUtility.cc.o CMakeFiles/ja2.dir/src/externalized/LoadingScreenModel.cc.o CMakeFiles/ja2.dir/src/externalized/MagazineModel.cc.o CMakeFiles/ja2.dir/src/externalized/ModPackContentManager.cc.o CMakeFiles/ja2.dir/src/externalized/RustInterface.cc.o CMakeFiles/ja2.dir/src/externalized/ShippingDestinationModel.cc.o CMakeFiles/ja2.dir/src/externalized/Soldier.cc.o CMakeFiles/ja2.dir/src/externalized/VehicleModel.cc.o CMakeFiles/ja2.dir/src/externalized/WeaponModels.cc.o CMakeFiles/ja2.dir/src/game/Editor/Cursor_Modes.cc.o CMakeFiles/ja2.dir/src/game/Editor/EditScreen.cc.o CMakeFiles/ja2.dir/src/game/Editor/Edit_Sys.cc.o CMakeFiles/ja2.dir/src/game/Editor/EditorBuildings.cc.o CMakeFiles/ja2.dir/src/game/Editor/EditorItems.cc.o CMakeFiles/ja2.dir/src/game/Editor/EditorMapInfo.cc.o CMakeFiles/ja2.dir/src/game/Editor/EditorMercs.cc.o CMakeFiles/ja2.dir/src/game/Editor/EditorTerrain.cc.o CMakeFiles/ja2.dir/src/game/Editor/Editor_Callbacks.cc.o CMakeFiles/ja2.dir/src/game/Editor/Editor_Modes.cc.o CMakeFiles/ja2.dir/src/game/Editor/Editor_Taskbar_Creation.cc.o CMakeFiles/ja2.dir/src/game/Editor/Editor_Taskbar_Utils.cc.o CMakeFiles/ja2.dir/src/game/Editor/Editor_Undo.cc.o CMakeFiles/ja2.dir/src/game/Editor/Item_Statistics.cc.o CMakeFiles/ja2.dir/src/game/Editor/LoadScreen.cc.o CMakeFiles/ja2.dir/src/game/Editor/MessageBox.cc.o CMakeFiles/ja2.dir/src/game/Editor/NewSmooth.cc.o CMakeFiles/ja2.dir/src/game/Editor/PopupMenu.cc.o CMakeFiles/ja2.dir/src/game/Editor/Road_Smoothing.cc.o CMakeFiles/ja2.dir/src/game/Editor/Sector_Summary.cc.o CMakeFiles/ja2.dir/src/game/Editor/SelectWin.cc.o CMakeFiles/ja2.dir/src/game/Editor/SmartMethod.cc.o CMakeFiles/ja2.dir/src/game/Editor/Smooth.cc.o CMakeFiles/ja2.dir/src/game/Editor/Smoothing_Utils.cc.o CMakeFiles/ja2.dir/src/game/Laptop/AIM.cc.o CMakeFiles/ja2.dir/src/game/Laptop/AIMArchives.cc.o CMakeFiles/ja2.dir/src/game/Laptop/AIMFacialIndex.cc.o CMakeFiles/ja2.dir/src/game/Laptop/AIMHistory.cc.o CMakeFiles/ja2.dir/src/game/Laptop/AIMLinks.cc.o CMakeFiles/ja2.dir/src/game/Laptop/AIMMembers.cc.o CMakeFiles/ja2.dir/src/game/Laptop/AIMPolicies.cc.o CMakeFiles/ja2.dir/src/game/Laptop/AIMSort.cc.o CMakeFiles/ja2.dir/src/game/Laptop/BobbyR.cc.o CMakeFiles/ja2.dir/src/game/Laptop/BobbyRAmmo.cc.o CMakeFiles/ja2.dir/src/game/Laptop/BobbyRArmour.cc.o CMakeFiles/ja2.dir/src/game/Laptop/BobbyRGuns.cc.o CMakeFiles/ja2.dir/src/game/Laptop/BobbyRMailOrder.cc.o CMakeFiles/ja2.dir/src/game/Laptop/BobbyRMisc.cc.o CMakeFiles/ja2.dir/src/game/Laptop/BobbyRShipments.cc.o CMakeFiles/ja2.dir/src/game/Laptop/BobbyRUsed.cc.o CMakeFiles/ja2.dir/src/game/Laptop/BrokenLink.cc.o CMakeFiles/ja2.dir/src/game/Laptop/CharProfile.cc.o CMakeFiles/ja2.dir/src/game/Laptop/EMail.cc.o CMakeFiles/ja2.dir/src/game/Laptop/Files.cc.o CMakeFiles/ja2.dir/src/game/Laptop/Finances.cc.o CMakeFiles/ja2.dir/src/game/Laptop/Florist.cc.o CMakeFiles/ja2.dir/src/game/Laptop/Florist_Cards.cc.o CMakeFiles/ja2.dir/src/game/Laptop/Florist_Gallery.cc.o CMakeFiles/ja2.dir/src/game/Laptop/Florist_Order_Form.cc.o CMakeFiles/ja2.dir/src/game/Laptop/Funeral.cc.o CMakeFiles/ja2.dir/src/game/Laptop/History.cc.o CMakeFiles/ja2.dir/src/game/Laptop/IMPVideoObjects.cc.o CMakeFiles/ja2.dir/src/game/Laptop/IMP_AboutUs.cc.o CMakeFiles/ja2.dir/src/game/Laptop/IMP_Attribute_Entrance.cc.o CMakeFiles/ja2.dir/src/game/Laptop/IMP_Attribute_Finish.cc.o CMakeFiles/ja2.dir/src/game/Laptop/IMP_Attribute_Selection.cc.o CMakeFiles/ja2.dir/src/game/Laptop/IMP_Begin_Screen.cc.o CMakeFiles/ja2.dir/src/game/Laptop/IMP_Compile_Character.cc.o CMakeFiles/ja2.dir/src/game/Laptop/IMP_Confirm.cc.o CMakeFiles/ja2.dir/src/game/Laptop/IMP_Finish.cc.o CMakeFiles/ja2.dir/src/game/Laptop/IMP_HomePage.cc.o CMakeFiles/ja2.dir/src/game/Laptop/IMP_MainPage.cc.o CMakeFiles/ja2.dir/src/game/Laptop/IMP_Personality_Entrance.cc.o CMakeFiles/ja2.dir/src/game/Laptop/IMP_Personality_Finish.cc.o CMakeFiles/ja2.dir/src/game/Laptop/IMP_Personality_Quiz.cc.o CMakeFiles/ja2.dir/src/game/Laptop/IMP_Portraits.cc.o CMakeFiles/ja2.dir/src/game/Laptop/IMP_SkillTraits.cc.o CMakeFiles/ja2.dir/src/game/Laptop/IMP_Text_System.cc.o CMakeFiles/ja2.dir/src/game/Laptop/IMP_Voices.cc.o CMakeFiles/ja2.dir/src/game/Laptop/Insurance.cc.o CMakeFiles/ja2.dir/src/game/Laptop/Insurance_Comments.cc.o CMakeFiles/ja2.dir/src/game/Laptop/Insurance_Contract.cc.o CMakeFiles/ja2.dir/src/game/Laptop/Insurance_Info.cc.o CMakeFiles/ja2.dir/src/game/Laptop/Laptop.cc.o CMakeFiles/ja2.dir/src/game/Laptop/Mercs.cc.o CMakeFiles/ja2.dir/src/game/Laptop/Mercs_Account.cc.o CMakeFiles/ja2.dir/src/game/Laptop/Mercs_Files.cc.o CMakeFiles/ja2.dir/src/game/Laptop/Mercs_No_Account.cc.o CMakeFiles/ja2.dir/src/game/Laptop/Personnel.cc.o CMakeFiles/ja2.dir/src/game/Laptop/Store_Inventory.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Assignments.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Auto_Resolve.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Campaign_Init.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Creature_Spreading.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Game_Clock.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Game_Event_Hook.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Game_Events.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Game_Init.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Hourly_Update.cc.o CMakeFiles/ja2.dir/src/game/Strategic/LoadSaveSectorInfo.cc.o CMakeFiles/ja2.dir/src/game/Strategic/LoadSaveStrategicMapElement.cc.o CMakeFiles/ja2.dir/src/game/Strategic/LoadSaveUndergroundSectorInfo.cc.o CMakeFiles/ja2.dir/src/game/Strategic/MapScreen.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Map_Screen_Helicopter.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Map_Screen_Interface.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Map_Screen_Interface_Border.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Map_Screen_Interface_Bottom.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Map_Screen_Interface_Map.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Map_Screen_Interface_Map_Inventory.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Map_Screen_Interface_TownMine_Info.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Meanwhile.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Merc_Contract.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Player_Command.cc.o CMakeFiles/ja2.dir/src/game/Strategic/PreBattle_Interface.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Queen_Command.cc.o CMakeFiles/ja2.dir/src/game/Strategic/QuestText.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Quest_Debug_System.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Quests.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Scheduling.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Strategic.cc.o CMakeFiles/ja2.dir/src/game/Strategic/StrategicMap.cc.o CMakeFiles/ja2.dir/src/game/Strategic/StrategicMap_Secrets.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Strategic_AI.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Strategic_Event_Handler.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Strategic_Merc_Handler.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Strategic_Mines.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Strategic_Movement.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Strategic_Movement_Costs.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Strategic_Pathing.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Strategic_Status.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Strategic_Town_Loyalty.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Strategic_Turns.cc.o CMakeFiles/ja2.dir/src/game/Strategic/SAM_Sites.cc.o CMakeFiles/ja2.dir/src/game/Strategic/Town_Militia.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Animation_Cache.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Animation_Control.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Animation_Data.cc.o CMakeFiles/ja2.dir/src/game/Tactical/ArmsDealerInvInit.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Arms_Dealer_Init.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Auto_Bandage.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Boxing.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Bullets.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Campaign.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Civ_Quotes.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Dialogue_Control.cc.o CMakeFiles/ja2.dir/src/game/Tactical/DisplayCover.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Drugs_And_Alcohol.cc.o CMakeFiles/ja2.dir/src/game/Tactical/End_Game.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Enemy_Soldier_Save.cc.o CMakeFiles/ja2.dir/src/game/Tactical/FOV.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Faces.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Gap.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Handle_Doors.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Handle_Items.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Handle_UI.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Interface.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Interface_Control.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Interface_Cursors.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Interface_Dialogue.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Interface_Items.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Interface_Panels.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Interface_Utils.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Inventory_Choosing.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Items.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Keys.cc.o CMakeFiles/ja2.dir/src/game/Tactical/LOS.cc.o CMakeFiles/ja2.dir/src/game/Tactical/LoadSaveBasicSoldierCreateStruct.cc.o CMakeFiles/ja2.dir/src/game/Tactical/LoadSaveBullet.cc.o CMakeFiles/ja2.dir/src/game/Tactical/LoadSaveMercProfile.cc.o CMakeFiles/ja2.dir/src/game/Tactical/LoadSaveObjectType.cc.o CMakeFiles/ja2.dir/src/game/Tactical/LoadSaveRottingCorpse.cc.o CMakeFiles/ja2.dir/src/game/Tactical/LoadSaveSoldierCreate.cc.o CMakeFiles/ja2.dir/src/game/Tactical/LoadSaveSoldierType.cc.o CMakeFiles/ja2.dir/src/game/Tactical/LoadSaveVehicleType.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Map_Information.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Merc_Entering.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Merc_Hiring.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Militia_Control.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Morale.cc.o CMakeFiles/ja2.dir/src/game/Tactical/OppList.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Overhead.cc.o CMakeFiles/ja2.dir/src/game/Tactical/PathAI.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Points.cc.o CMakeFiles/ja2.dir/src/game/Tactical/QArray.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Real_Time_Input.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Rotting_Corpses.cc.o CMakeFiles/ja2.dir/src/game/Tactical/ShopKeeper_Interface.cc.o CMakeFiles/ja2.dir/src/game/Tactical/SkillCheck.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Soldier_Add.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Soldier_Ani.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Soldier_Control.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Soldier_Create.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Soldier_Find.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Soldier_Init_List.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Soldier_Profile.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Soldier_Tile.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Spread_Burst.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Squads.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Strategic_Exit_GUI.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Structure_Wrap.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Tactical_Save.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Tactical_Turns.cc.o CMakeFiles/ja2.dir/src/game/Tactical/TeamTurns.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Touch_UI.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Turn_Based_Input.cc.o CMakeFiles/ja2.dir/src/game/Tactical/UI_Cursors.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Vehicles.cc.o CMakeFiles/ja2.dir/src/game/Tactical/Weapons.cc.o CMakeFiles/ja2.dir/src/game/Tactical/World_Items.cc.o CMakeFiles/ja2.dir/src/game/TacticalAI/AIList.cc.o CMakeFiles/ja2.dir/src/game/TacticalAI/AIMain.cc.o CMakeFiles/ja2.dir/src/game/TacticalAI/AIUtils.cc.o CMakeFiles/ja2.dir/src/game/TacticalAI/Attacks.cc.o CMakeFiles/ja2.dir/src/game/TacticalAI/CreatureDecideAction.cc.o CMakeFiles/ja2.dir/src/game/TacticalAI/DecideAction.cc.o CMakeFiles/ja2.dir/src/game/TacticalAI/FindLocations.cc.o CMakeFiles/ja2.dir/src/game/TacticalAI/Knowledge.cc.o CMakeFiles/ja2.dir/src/game/TacticalAI/Medical.cc.o CMakeFiles/ja2.dir/src/game/TacticalAI/Movement.cc.o CMakeFiles/ja2.dir/src/game/TacticalAI/NPC.cc.o CMakeFiles/ja2.dir/src/game/TacticalAI/PanicButtons.cc.o CMakeFiles/ja2.dir/src/game/TacticalAI/Realtime.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/Ambient_Control.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/Buildings.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/Environment.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/Exit_Grids.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/Explosion_Control.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/Fog_Of_War.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/Interactive_Tiles.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/Isometric_Utils.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/LightEffects.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/Lighting.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/LoadSaveExplosionType.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/LoadSaveLightEffect.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/LoadSaveLightSprite.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/LoadSaveRealObject.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/LoadSaveSmokeEffect.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/Map_Edgepoints.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/Overhead_Map.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/Phys_Math.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/Physics.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/Pits.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/Radar_Screen.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/RenderWorld.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/Render_Dirty.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/Render_Fun.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/SaveLoadMap.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/Simple_Render_Utils.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/Smell.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/SmokeEffects.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/Structure.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/SysUtil.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/Tactical_Placement_GUI.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/TileDat.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/TileDef.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/Tile_Animation.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/Tile_Cache.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/Tile_Surface.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/WorldDat.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/WorldDef.cc.o CMakeFiles/ja2.dir/src/game/TileEngine/WorldMan.cc.o CMakeFiles/ja2.dir/src/game/Utils/Animated_ProgressBar.cc.o CMakeFiles/ja2.dir/src/game/Utils/Cinematics.cc.o CMakeFiles/ja2.dir/src/game/Utils/Cursors.cc.o CMakeFiles/ja2.dir/src/game/Utils/Debug_Pages.cc.o CMakeFiles/ja2.dir/src/game/Utils/Event_Pump.cc.o CMakeFiles/ja2.dir/src/game/Utils/Font_Control.cc.o CMakeFiles/ja2.dir/src/game/Utils/MercTextBox.cc.o CMakeFiles/ja2.dir/src/game/Utils/Message.cc.o CMakeFiles/ja2.dir/src/game/Utils/Music_Control.cc.o CMakeFiles/ja2.dir/src/game/Utils/Observable.cc.o CMakeFiles/ja2.dir/src/game/Utils/PopUpBox.cc.o CMakeFiles/ja2.dir/src/game/Utils/STIConvert.cc.o CMakeFiles/ja2.dir/src/game/Utils/Slider.cc.o CMakeFiles/ja2.dir/src/game/Utils/Sound_Control.cc.o CMakeFiles/ja2.dir/src/game/Utils/Text_Input.cc.o CMakeFiles/ja2.dir/src/game/Utils/Text_Utils.cc.o CMakeFiles/ja2.dir/src/game/Utils/Timer_Control.cc.o CMakeFiles/ja2.dir/src/game/Utils/Utilities.cc.o CMakeFiles/ja2.dir/src/game/Utils/WordWrap.cc.o CMakeFiles/ja2.dir/src/game/Credits.cc.o CMakeFiles/ja2.dir/src/game/UILayout.cc.o CMakeFiles/ja2.dir/src/game/Fade_Screen.cc.o CMakeFiles/ja2.dir/src/game/GameInitOptionsScreen.cc.o CMakeFiles/ja2.dir/src/game/GameLoop.cc.o CMakeFiles/ja2.dir/src/game/GameRes.cc.o CMakeFiles/ja2.dir/src/game/GameMode.cc.o CMakeFiles/ja2.dir/src/game/GameScreen.cc.o CMakeFiles/ja2.dir/src/game/GameSettings.cc.o CMakeFiles/ja2.dir/src/game/GameVersion.cc.o CMakeFiles/ja2.dir/src/game/HelpScreen.cc.o CMakeFiles/ja2.dir/src/game/Init.cc.o CMakeFiles/ja2.dir/src/game/Intro.cc.o CMakeFiles/ja2.dir/src/game/JA2_Splash.cc.o CMakeFiles/ja2.dir/src/game/JAScreens.cc.o CMakeFiles/ja2.dir/src/game/LoadSaveEMail.cc.o CMakeFiles/ja2.dir/src/game/LoadSaveTacticalStatusType.cc.o CMakeFiles/ja2.dir/src/game/Loading_Screen.cc.o CMakeFiles/ja2.dir/src/game/MainMenuScreen.cc.o CMakeFiles/ja2.dir/src/game/MercPortrait.cc.o CMakeFiles/ja2.dir/src/game/MessageBoxScreen.cc.o CMakeFiles/ja2.dir/src/game/Options_Screen.cc.o CMakeFiles/ja2.dir/src/game/SaveLoadGame.cc.o CMakeFiles/ja2.dir/src/game/SaveLoadGameStates.cc.o CMakeFiles/ja2.dir/src/game/SaveLoadScreen.cc.o CMakeFiles/ja2.dir/src/game/Screens.cc.o CMakeFiles/ja2.dir/src/sgp/Button_Sound_Control.cc.o CMakeFiles/ja2.dir/src/sgp/Button_System.cc.o CMakeFiles/ja2.dir/src/sgp/Cursor_Control.cc.o CMakeFiles/ja2.dir/src/sgp/DirFs.cc.o CMakeFiles/ja2.dir/src/sgp/EncodingCorrectors.cc.o CMakeFiles/ja2.dir/src/sgp/FileMan.cc.o CMakeFiles/ja2.dir/src/sgp/Font.cc.o CMakeFiles/ja2.dir/src/sgp/FPS.cc.o CMakeFiles/ja2.dir/src/sgp/HImage.cc.o CMakeFiles/ja2.dir/src/sgp/ImpTGA.cc.o CMakeFiles/ja2.dir/src/sgp/Input.cc.o CMakeFiles/ja2.dir/src/sgp/Line.cc.o CMakeFiles/ja2.dir/src/sgp/LoadSaveData.cc.o CMakeFiles/ja2.dir/src/sgp/MouseSystem.cc.o CMakeFiles/ja2.dir/src/sgp/PCX.cc.o CMakeFiles/ja2.dir/src/sgp/Random.cc.o CMakeFiles/ja2.dir/src/sgp/SGP.cc.o CMakeFiles/ja2.dir/src/sgp/SGPFile.cc.o CMakeFiles/ja2.dir/src/sgp/SGPStrings.cc.o CMakeFiles/ja2.dir/src/sgp/STCI.cc.o CMakeFiles/ja2.dir/src/sgp/Shading.cc.o CMakeFiles/ja2.dir/src/sgp/SoundMan.cc.o CMakeFiles/ja2.dir/src/sgp/Types.cc.o CMakeFiles/ja2.dir/src/sgp/VObject.cc.o CMakeFiles/ja2.dir/src/sgp/VObject_Blitters.cc.o CMakeFiles/ja2.dir/src/sgp/VSurface.cc.o CMakeFiles/ja2.dir/src/sgp/Video.cc.o -o ja2  /usr/lib64/libSDL2main.a  /usr/lib64/libSDL2.so  dependencies/lib-smacker/libsmacker.a  lib-stracciatella/lib/libstracciatella.a  -ldl  -lpthread  /usr/lib64/liblua5.3.so && cd /var/tmp/portage/games-strategy/ja2-stracciatella-0.21.0/work/ja2-stracciatella-0.21.0_build && /usr/bin/cmake -E copy_directory /var/tmp/portage/games-strategy/ja2-stracciatella-0.21.0/work/ja2-stracciatella-0.21.0/assets/externalized /var/tmp/portage/games-strategy/ja2-stracciatella-0.21.0/work/ja2-stracciatella-0.21.0_build/externalized && cd /var/tmp/portage/games-strategy/ja2-stracciatella-0.21.0/work/ja2-stracciatella-0.21.0_build && /usr/bin/cmake -E copy_directory /var/tmp/portage/games-strategy/ja2-stracciatella-0.21.0/work/ja2-stracciatella-0.21.0/assets/unittests /var/tmp/portage/games-strategy/ja2-stracciatella-0.21.0/work/ja2-stracciatella-0.21.0_build/unittests && cd /var/tmp/portage/games-strategy/ja2-stracciatella-0.21.0/work/ja2-stracciatella-0.21.0_build && /usr/bin/cmake -E copy_directory /var/tmp/portage/games-strategy/ja2-stracciatella-0.21.0/work/ja2-stracciatella-0.21.0/assets/mods /var/tmp/portage/games-strategy/ja2-stracciatella-0.21.0/work/ja2-stracciatella-0.21.0_build/mods
/var/tmp/portage/games-strategy/ja2-stracciatella-0.21.0/work/ja2-stracciatella-0.21.0/src/game/Strategic/Strategic_Pathing.cc:27:8: error: type ‘struct path_t’ violates the C++ One Definition Rule [-Werror=odr]
   27 | struct path_t
      |        ^
/var/tmp/portage/games-strategy/ja2-stracciatella-0.21.0/work/ja2-stracciatella-0.21.0/src/game/Tactical/PathAI.cc:80:8: note: a different type is defined in another translation unit
   80 | struct path_t
      |        ^
/var/tmp/portage/games-strategy/ja2-stracciatella-0.21.0/work/ja2-stracciatella-0.21.0/src/game/Strategic/Strategic_Pathing.cc:29:15: note: the first difference of corresponding definitions is field ‘nextLink’
   29 |         INT16 nextLink;           //2
      |               ^
/var/tmp/portage/games-strategy/ja2-stracciatella-0.21.0/work/ja2-stracciatella-0.21.0/src/game/Tactical/PathAI.cc:82:17: note: a field with different name is defined in another translation unit
   82 |         INT32   iLocation; //4
      |                 ^
/var/tmp/portage/games-strategy/ja2-stracciatella-0.21.0/work/ja2-stracciatella-0.21.0/src/game/Strategic/Strategic_Pathing.cc:37:8: error: type ‘struct trail_t’ violates the C++ One Definition Rule [-Werror=odr]
   37 | struct trail_t
      |        ^
/var/tmp/portage/games-strategy/ja2-stracciatella-0.21.0/work/ja2-stracciatella-0.21.0/src/game/Tactical/PathAI.cc:93:8: note: a different type is defined in another translation unit
   93 | struct trail_t
      |        ^
/var/tmp/portage/games-strategy/ja2-stracciatella-0.21.0/work/ja2-stracciatella-0.21.0/src/game/Strategic/Strategic_Pathing.cc:40:15: note: the first difference of corresponding definitions is field ‘diStratDelta’
   40 |         short diStratDelta;
      |               ^
/var/tmp/portage/games-strategy/ja2-stracciatella-0.21.0/work/ja2-stracciatella-0.21.0/src/game/Tactical/PathAI.cc:96:15: note: a field with different name is defined in another translation unit
   96 |         UINT8 stepDir;
      |               ^
lto1: some warnings being treated as errors
lto-wrapper: fatal error: /usr/bin/x86_64-pc-linux-gnu-g++ returned 1 exit status
compilation terminated.
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: error: lto-wrapper failed
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

Previously reported downstream: https://bugs.gentoo.org/916965
Full build log: [build.log](https://github.com/ja2-stracciatella/ja2-stracciatella/files/14805737/build.log)

--- a/src/game/Strategic/Strategic_Pathing.cc
+++ b/src/game/Strategic/Strategic_Pathing.cc
@@ -24,6 +24,15 @@ static BOOLEAN gfPlotToAvoidPlayerInfuencedSectors = FALSE;
 
 
 // Globals
+
+namespace StrategicPathing
+{
+// These type definitions were placed inside a namespace to avoid a collision
+// with their counterparts in Tactical/PathAI.cc which caused a violation of
+// the C++ One Definition Rule, issue #1963.
+
+using TRAILCELLTYPE = UINT32;
+
 struct path_t
 {
 	INT16 nextLink;           //2
@@ -40,6 +49,9 @@ struct trail_t
 	short diStratDelta;
 };
 
+}
+using namespace StrategicPathing;
+
 #define MAXTRAILTREE	(4096)
 #define MAXpathQ			(512)
 #define MAP_WIDTH 18
@@ -49,7 +61,6 @@ struct trail_t
 //#define ISWATER(t)	(((t)==TRAVELCOST_KNEEDEEP) || ((t)==TRAVELCOST_DEEPWATER))
 //#define NOPASS (TRAVELCOST_OBSTACLE)
 //#define VEINCOST TRAVELCOST_FLAT     //actual cost for bridges and doors and such
-#define TRAILCELLTYPE UINT32
 #define TRAILCELLMAX 0xFFFFFFFF
 
 static path_t        pathQB[MAXpathQ];

--- a/src/game/Strategic/Strategic_Pathing.h
+++ b/src/game/Strategic/Strategic_Pathing.h
@@ -13,18 +13,6 @@
 #define SOUTH_MOVE  18
 
 
-// Movement speed defines
-#define NORMAL_MVT 1
-#define SLOW_MVT   0
-
-
-// movment modes
-enum{
-	MVT_MODE_AIR,
-	MVT_MODE_VEHICLE,
-	MVT_MODE_FOOT,
-};
-
 INT32 FindStratPath(INT16 sStart, INT16 sDestination, GROUP const&, BOOLEAN fTacticalTraversal);
 
 // build a stategic path

